MIGRATING.md

```diff
@@ -0,0 +1,57 @@
+
+### "Active" token renew
+
+Previously, tokens would only be renewed when they were read from storge. This typically occurred when a user was navigating to a protected route. Now, tokens will be renewed in the background before they expire. If token renew fails, the [authState](https://github.com/okta/okta-auth-js/blob/master/README.md#authstatemanager) will be updated and `authState.isAuthenticated` will be recalculated. If the user is currently on a protected route, they will need to re-authenticate. Set the `onAuthRequired` option to customize behavior when authentication is required. You can set [tokenManager.autoRenew](https://github.com/okta/okta-auth-js/blob/master/README.md#autorenew) to `false` to disable active token renew logic.
+
+### `login` is removed
+
+This method called `onAuthRequired`, if it was set in the config options, or `loginRedirect` if no `onAuthRequired` option was set. If you had code that wa calling this method, you may either call your `onAuthRequired` function directly or `signInWithRedirect`.
+
+### `loginRedirect` is replaced by `signInWithRedirect`
+
+`loginRedirect` took 2 parameters: a `fromUri` and `additionalParams`. The replacement method, [signInWithRedirect](https://github.com/okta/okta-auth-js/blob/master/README.md#signinwithredirectoptions) takes only one argument, called `options` which can include a value for `originalUri` which is equivalent to `fromUri`. It is the URL which will be set after the login flow is complete. Other options which were previously set on `additionalParams` can also be set on `options`.
+
+If you had code like this:
+
+```javascript
+okta.loginRedirect('/profile', { scopes: ['openid', 'profile'] });
+```
+
+it can be rewritten as:
+
+```javascript
+okta.signInWithRedirect({ originalUri: '/profile', scopes: ['openid', 'profile'] });
+```
+
+### `logout` is replaced by `signOut`
+
+`logout` accepted either a string or an object as options. [signOut](https://github.com/okta/okta-auth-js/blob/master/README.md#signout) accepts only an options object.
+
+If you had code like this:
+
+```javascript
+okta.logout('/goodbye');
+```
+
+it can be rewritten as:
+
+```javascript
+okta.signOut({ postLogoutRedirectUri: window.location.orign + '/goodbye' });
+```
+
+Note that the value for `postLogoutRedirectUri` must be an absolute URL. This URL must also be on the "allowed list" in your Okta app's configuration. If no options are passed or no `postLogoutRedirectUri` is set on the options object, it will redirect to `window.location.origin` after sign out is complete.
+
+### `handleAuthentication` is replaced by `handleLoginRedirect`
+
+`handleLoginRedirect` is called by the `OktaLoginCallback` component as the last step of the login redirect authorization flow. It will obtain and store tokens and then restore the `originalUri` which was set before the login redirect flow began.
+
+### `isAuthenticated` is called by `transformAuthState`
```

Comment:
why not just deprecate `isAuthenticated` callback, then suggest users start to use the `transformAuthState `? The previous common use case for `isAuthenticated` is to override the evaluation to use `idToken && accessToken` which has been set to default in the current breaking change. Then leave both options here probably won't help too much.

Reply:
That was my initial thought. I changed my mind because `isAuthenticated` passes an instance of `okta-angular` whereas `transformAuthState` passes an instance of `OktaAuth` (the base class). This isn't really a deal-breaker, if we really wanted it to receive an instance of `okta-angular` we could wrap the function and pass `this` which can be cast to either type. But it all seemed kind of complicated and advanced compared to what is done currently. I am open to removing isAuthenticated and just showing how to use `transformAuthState` but we should probably explain that this is for advanced custom cases only. If your current implementation of `isAuthenticated` matches the default logic, you should just remove it.